### PR TITLE
Feature/Client Books TimeSlots for Pets

### PR DESCRIPTION
**Pull request by loganSIO**
The actual branch name is not explicit, but I changed the title.
This PR refers to this issue : #https://github.com/MonChenil/monchenil/issues/11

### CODE REVIEW

File: backend/MonChenil/Services/TimeSlotService.cs [hunk 1]
```diff
@@ -58,9 +64,67 @@ private void CheckTimeSlot(TimeSlot timeSlot)
...
+                throw new ArgumentException("Incompatible pets cannot be added to the same time slot.", nameof(pets));
+            }
+        }
+
+        timeSlot.Pets.AddRange(pets);
+
+        _repository.Update(timeSlot);
+    }
+
+    private bool ArePetsIncompatible(Pet pet1, Pet pet2)
```
Comment (samymsa) [inline backend/MonChenil/Services/TimeSlotService.cs:120]: Rename to ArePetsCompatible()
Change calls from ArePetsIncompatible() to !ArePetsCompatible()

File: backend/MonChenil/Services/TimeSlotService.cs [hunk 2]
```diff
@@ -58,9 +64,67 @@ private void CheckTimeSlot(TimeSlot timeSlot)
...
+    {
+        return (pet1.Type == PetType.Cat && pet2.Type == PetType.Dog) ||
```
Comment (samymsa) [inline backend/MonChenil/Services/TimeSlotService.cs:122]: Update the logic :
Pets are compatible if :
- pet1.IncompatibleTypes does not contain pet2.Type
AND
- pet2.IncompatibleTypes does not contain pet1.Type

File: backend/MonChenil/Services/TimeSlotService.cs [hunk 3]
```diff
@@ -58,9 +64,67 @@ private void CheckTimeSlot(TimeSlot timeSlot)
...
+        }
+
+        timeSlot.Pets.AddRange(pets);
+
+        _repository.Update(timeSlot);
+    }
+
+    public void AddPetToTimeSlot(TimeSlot timeSlot, IEnumerable<Pet> pets)
+    {
+        if (timeSlot == null)
```
Comment (samymsa) [inline backend/MonChenil/Services/TimeSlotService.cs:97]: timeSlot cannot be null. Remove this unnecessary check

File: backend/MonChenil/Services/TimeSlotService.cs [hunk 4]
```diff
@@ -58,9 +64,67 @@ private void CheckTimeSlot(TimeSlot timeSlot)
...
+        {
+            throw new ArgumentNullException(nameof(timeSlot));
+        }
+
+        if (pets == null || !pets.Any())
```
Comment (samymsa) [inline backend/MonChenil/Services/TimeSlotService.cs:102]: pets cannot be null. Remove this unnecessary check. You can leave the pets.Any() check

File: backend/MonChenil/Services/TimeSlotService.cs [hunk 5]
```diff
@@ -58,9 +64,67 @@ private void CheckTimeSlot(TimeSlot timeSlot)
...
+                }
+            }
+        }
+
+        timeSlot.Pets.AddRange(pets);
+
+        _repository.Update(timeSlot);
+    }
+
+    public void AddPetToTimeSlot(TimeSlot timeSlot, IEnumerable<Pet> pets)
```
Comment (samymsa) [inline backend/MonChenil/Services/TimeSlotService.cs:95]: Rename to AddPetsToTimeSlot (plural)

File: backend/MonChenil/Services/TimeSlotService.cs [hunk 6]
```diff
@@ -58,9 +64,67 @@ private void CheckTimeSlot(TimeSlot timeSlot)
         }
     }
 
+    public void BookTimeSlotForPets(TimeSlot timeSlot, IEnumerable<Pet> pets)
```
Comment (samymsa) [inline backend/MonChenil/Services/TimeSlotService.cs:67]: This looks like a duplicate of AddPetsToTimeSlot()

File: backend/MonChenil/Services/TimeSlotService.cs [hunk 7]
```diff
@@ -24,6 +24,12 @@ public IEnumerable<TimeSlot> GetAll()
         return _repository.GetAll();
     }
 
+    public IEnumerable<TimeSlot> GetAvailableTimeSlots()
```
Comment (samymsa) [inline backend/MonChenil/Services/TimeSlotService.cs:27]: Take a list of pets as parameter
Filter the list of TimeSlots: Only return time slots that can be booked for the given pets. In other words, only return time slots for which there is no compatibility conflict with the given pets.

File: backend/MonChenil/Entities/TimeSlot.cs [hunk 8]
```diff
@@ -5,5 +5,5 @@ public class TimeSlot : IEntity
     public int Id { get; set; }
     public DateTime StartDate { get; set; }
     public DateTime EndDate { get; set; }
-    public List<Pet> Pets { get; set; } = [];
```
Comment (samymsa) [inline backend/MonChenil/Entities/TimeSlot.cs:8]: Why?

File: backend/MonChenil/Entities/Pet.cs [hunk 9]
```diff
@@ -14,4 +14,16 @@ public class Pet : IEntity
     public User? Owner { get; set; }
     public List<TimeSlot> TimeSlots { get; set; } = [];
     public List<PetType> IncompatibleTypes { get; set; } = [];
+
+    public Pet()
```
Comment (samymsa) [inline backend/MonChenil/Entities/Pet.cs:18]: Don't initialize incompatible pets. Let the user specify that.

**Review by samymsa — CHANGES_REQUESTED** [review]
Nice 1st PR.

Check the comments and update your tests accordingly 😉

**Comment by loganSIO** [comment]
Every modification requests are made in this PR #https://github.com/MonChenil/monchenil/pull/23

**Pull request closed without merging**